Sources/Casbin/Model/DefaultModel.swift

```diff
@@ -204,13 +204,15 @@ extension DefaultModel:Model {
     public func hasPolicy(sec:String,ptype:String,rule:[String]) -> Bool {
         getPolicy(sec: sec, ptype: ptype).contains(rule)
     }
-    public func getValuesForFieldInPolicy(sec:String,ptype:String,fieldIndex:Int) -> [String] {
-        getPolicy(sec: sec, ptype: ptype).reduce([]) { acc, x in
-            var acc = acc
-            acc.append(x[fieldIndex])
-            return acc
-        }
-    }
+    public func getValuesForFieldInPolicy(sec:String,ptype:String,fieldIndex:Int) -> [String]  {
+          let policy = getPolicy(sec: sec, ptype: ptype).reduce(Set<String>()) { acc, x in
+          var acc = acc
+          acc.insert(x[fieldIndex])
+          return acc
+      }
```

Comment:
Not only lines of code that `reduce(into:_:)` can reduce, it also can prevent copying set during each iteration. You know, when set is large, the copying operation will be time consuming. Although the set here may be quite small, I insist a method can avoid unnecessary copying should be better.

```suggestion
      let policy = getPolicy(sec: sec, ptype: ptype).reduce(into: Set<String>()) { acc, x in
          acc.insert(x[fieldIndex])
      }
```

Reply:
Thanks,I will take it to the code.

Reply:
This kind of writing should be reported. acc  is a 'let' constant